### ISSUE TITLE
[Bug]: clone does not handle slashes at the end of repository path.

### ISSUE THREAD

**Issue by spectre10** [gittuf/gittuf#202]
### What happened?

the Clone func added in #134 does not handle slashes at the end of repository path. 
```go
func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Repository, error) {
	if dir == "" {
		// FIXME: my understanding is backslashes are not used in URLs but I haven't dived into the RFCs to check yet
		split := strings.Split(strings.TrimSpace(strings.ReplaceAll(remoteURL, "\\", "/")), "/")
		dir = strings.TrimSuffix(split[len(split)-1], ".git")
	}
	..
	..
	if err := os.Mkdir(dir, 0755); err != nil {
		return nil, errors.Join(ErrCloningRepository, err)
	}
	..
}
```
Here, if we provide slashes at the end of remoteURL, split[len(split)-1] results in an empty string. Consequently, dir also results in an empty string. Afterwards, when dir is provided to Mkdir, it results in an error.

### Relevant log output

```Shell
$ ./gittuf clone https://github.com/gittuf/ci-demo.git/
Error: unable to clone repository
mkdir : no such file or directory
Usage:
  gittuf clone [flags]

Flags:
  -b, --branch string   Specify branch to check out
  -h, --help            help for clone
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by spectre10** [comment]
might be relevent #140

**Comment by adityasaky** [comment]
This is simpler than #140. Can you submit a patch to strip trailing slashes?

**Comment by adityasaky** [comment]
I'd probably strip it for the dirname rather than the URL as a whole first, and let go-git's URL clean up handle it after

**Comment by spectre10** [comment]
> This is simpler than #140. Can you submit a patch to strip trailing slashes?

Sure. #203

**Comment by adityasaky** [comment]
Fixed in #203
